### PR TITLE
[pkg/stanza] Add `send_quiet`/`drop_quiet` options for on_error setting

### PR DESCRIPTION
**Pull request by ChrsMark**
**Description:** <Describe what has changed.>

When stanza operators fail to process a message, no matter if they send the message or drop it an error message is logged. This can be quite "noisy" when users want to set up a best effort operator which on error should only send/drop the message without logging an error.
With introducing the `send_quiet` and `drop_quiet` options we provide the option to set the level of the logged error. `send_quiet` and `drop_quiet` log the error on debug level.

**Link to tracking Issue:** https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/32145

**Testing:** <Describe what testing was performed and which tests were added.> Added [unit-tests](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32220/files#diff-6a03331b322fcfd255ead096ac5c7c9dd9267a2d6697ea629548f37f4049896aR70)

**Documentation:** <Describe the documentation added.> [Enhanced the operator's docs](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/32220/files#diff-8f730eb2d61d9ca8c5cf863e9e7f3ddfe51984c46c6259dd7258902d6cc10090L2)

### CODE REVIEW

File: pkg/stanza/docs/types/on_error.md [hunk 1]
```diff
@@ -1,10 +1,21 @@
 # `on_error` parameter
-The `on_error` parameter determines the error handling strategy an operator should use when it fails to process an entry. There are 2 supported values: `drop` and `send`.
+The `on_error` parameter determines the error handling strategy an operator should use when it fails to
+process an entry. There are 2 supported values: `drop` and `send`.
```
Comment (OverOrion) [inline pkg/stanza/docs/types/on_error.md:3]:
```suggestion
process an entry. There are 4 supported values: `drop`, `send`, `send_quiet`, and `drop_quiet`.
```

File: pkg/stanza/docs/types/on_error.md [hunk 2]
```diff
@@ -1,10 +1,21 @@
...
-In this mode, if an operator fails to process an entry, it will still send the entry down the pipeline. This may result in downstream operators receiving entries in an undesired format.
\ No newline at end of file
+In this mode, if an operator fails to process an entry, it will still send the entry down the pipeline.
+This may result in downstream operators receiving entries in an undesired format.
+
+### `send_quiet`
+Same as `send` with only difference that the failure will be logged in debug level. Useful, when best effort
+operators are defined which might flood the logs with errors.
+
+### `drop_quiet`
```
Comment (OverOrion) [inline pkg/stanza/docs/types/on_error.md:19]: [nit]
How about the following order?
- `drop`
- `drop_quiet`
- `send`
- `send_quiet`

**Review by OverOrion — COMMENTED** [review]
Overall LGTM, only the docs part need minor adjustments.

File: pkg/stanza/docs/types/on_error.md [hunk 3]
```diff
@@ -1,10 +1,21 @@
 # `on_error` parameter
-The `on_error` parameter determines the error handling strategy an operator should use when it fails to process an entry. There are 2 supported values: `drop` and `send`.
+The `on_error` parameter determines the error handling strategy an operator should use when it fails to
+process an entry. There are 2 supported values: `drop`, `drop_quiet`, `send` and `send_quiet`.
```
Comment (djaglowski) [inline pkg/stanza/docs/types/on_error.md:3]:
```suggestion
process an entry. There are 4 supported values: `drop`, `drop_quiet`, `send` and `send_quiet`.
```

**Review by djaglowski — APPROVED** [review]

**Pull request merged**